src/gui/screen_lan_settings.c

```diff
@@ -72,16 +72,19 @@ static void _get_ip4_addrs(void) {
 }
 
 static void _addrs_to_str(char *param_str, uint8_t flg) {
-    static char ip4_addr_str[IP4_ADDR_STR_SIZE], ip4_msk_str[IP4_ADDR_STR_SIZE], ip4_gw_str[IP4_ADDR_STR_SIZE];
+    static char ip4_addr_str[IP4_ADDR_STR_SIZE], ip4_msk_str[IP4_ADDR_STR_SIZE], ip4_gw_str[IP4_ADDR_STR_SIZE], ip4_connect_str[IP4_ADDR_STR_SIZE];
     strlcpy(ip4_addr_str, ip4addr_ntoa(&(config.lan_ip4_addr)), IP4_ADDR_STR_SIZE);
     strlcpy(ip4_msk_str, ip4addr_ntoa(&(config.lan_ip4_msk)), IP4_ADDR_STR_SIZE);
     strlcpy(ip4_gw_str, ip4addr_ntoa(&(config.lan_ip4_gw)), IP4_ADDR_STR_SIZE);
 
     if (flg) {
-        char save_hostname[LAN_HOSTNAME_MAX_LEN + 1];
-        eeprom_get_hostname(save_hostname);
-        snprintf(param_str, MAX_INI_SIZE, "[lan_ip4]\ntype=%s\nhostname=%s\naddress=%s\nmask=%s\ngateway=%s",
-            config.lan_flag & LAN_EEFLG_TYPE ? LAN_type_opt[1] : LAN_type_opt[0], save_hostname, ip4_addr_str, ip4_msk_str, ip4_gw_str);
+        char save_hostname[LAN_HOSTNAME_MAX_LEN + 1], save_sec_key[CONNECT_SEC_KEY_LEN + 1];
```

Comment:
Loading and saving an INI file inside the screen_lan_settings.c is not a good practice
A proper solution would be to create a serialization "object", that can load/save/notify about the config changes.
A UI screen has nothing to do with some file saving...

Reply:
how much time do we have to fix this?